src/GraphQL/Types/Scalars/UriGraphType.cs

```diff
@@ -10,13 +10,29 @@ namespace GraphQL.Types
     public class UriGraphType : ScalarGraphType
     {
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value)
-            => value is StringValue stringValue ? ParseValue(stringValue.Value) : null;
+        public override object ParseLiteral(IValue value) => value switch
+        {
+            StringValue s => new Uri(s.Value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(Uri));
+        public override object ParseValue(object value) => value switch
+        {
+            string s => new Uri(s),
+            Uri _ => value,
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new StringValue(ParseValue(value).ToString());
+        public override object Serialize(object value) => value switch
+        {
+            string s => new Uri(s).ToString(), //validate uri
+            Uri u => u.ToString(),
+            null => null,
+            _ => ThrowSerializationError(value)
+        };
```

Comment:
```suggestion
        public override object Serialize(object value) => ParseValue(value)?.ToString();
```

The same as for `IdGraphType`.

Reply:
This cannot be done here.  ParseValue converts strings to Uris.  Serialize converts Uris to strings.

Reply:
Yes, the new code does it, nothing changes.

Reply:
Serialize: string or uri -> uri -> string

Reply:
True.

Done.